pkg/cloudencrypt/azure.go

```diff
@@ -0,0 +1,75 @@
+package cloudencrypt
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azkeys"
+)
+
+// AzureEncryptor Implements Encryptor by saving the value as a secret in Azure's Key Vault.
+type AzureEncryptor struct {
+	client  *azkeys.Client
+	keyName string
+}
+
+func NewAzureEncryptor(ctx context.Context, vaultBaseURL, keyName string) (*AzureEncryptor, error) {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := azkeys.NewClient(vaultBaseURL, cred, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &AzureEncryptor{
+		client:  client,
+		keyName: keyName,
+	}, nil
+}
+
+func (encryptor *AzureEncryptor) Encrypt(ctx context.Context, plaintext []byte, metadata ...MetadataKV) ([]byte, error) {
+	algorithm := azkeys.EncryptionAlgorithmRSAOAEP256
+
+	result, err := encryptor.client.Encrypt(
+		ctx,
+		encryptor.keyName,
+		"",
```

Comment:
Why key version is always empty? I would expect it is configurable or somehow you can change it except of rebuilding whole library.

Reply:
Empty means "use latest".

Reply:
I could swear it was stated in a comment in the code somewhere... can't find it now though.

I don't think the version is useful for our use case at all. You'd always need to know which version was used to encrypt the value in order to decrypt it. And there are no versions on the other clouds.

Reply:
I'll resolve it but we wanted to do a library from it, so it will be missing.